portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java

```diff
@@ -34,13 +34,20 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query();
-
+    String commentId = request.getParameter("comment-id");
+    Query query = new Query("Comment");
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
-
-    for (Entity entity : results.asIterable()) {
+    if (commentId.equals("all")) {
+      for (Entity entity : results.asIterable()) {
         datastore.delete(entity.getKey());
+      }
+    } else {
+      for (Entity entity : results.asIterable()) {
+        if (entity.getProperty("id").equals(commentId)){
+          datastore.delete(entity.getKey());
+        }
+      }
```

Comment:
Fetching all of the comment records and iterating through them to find a match to delete is going to be a very expensive operation. You should consider replacing this with a DELETE query instead, if possible.

Reply:
I wasn't quite sure how to implement a delete query, but I added a filter so that the query only fetches the entities with the matching id, so that I'm not iterating over every comment entity. Let me know if this isn't an adequate improvement!

Reply:
That's fine for now, but if you figure out how to do this further on feel free to come back around and improve this solution